gui/src/installer/step/mod.rs

```diff
@@ -229,6 +638,71 @@ impl From<DefineBitcoind> for Box<dyn Step> {
     }
 }
 
+impl From<StartInternalBitcoindStep> for Box<dyn Step> {
+    fn from(s: StartInternalBitcoindStep) -> Box<dyn Step> {
+        Box::new(s)
+    }
+}
+
+impl StartInternalBitcoindStep {
+    pub fn new(liana_datadir: &PathBuf) -> Self {
+        Self {
+            conf_path: internal_bitcoind_config_path(liana_datadir),
+            network: Network::Bitcoin,
+            started: None,
+            exe_path: form::Value::default(),
+        }
+    }
+}
+
+impl Step for StartInternalBitcoindStep {
+    fn load_context(&mut self, ctx: &Context) {
+        if self.exe_path.value.is_empty() {
+            self.exe_path.value = bitcoind_exe_path();
+        }
+        self.network = ctx.bitcoin_config.network;
+    }
+    fn update(&mut self, message: Message) -> Command<Message> {
+        if let Message::StartInternalBitcoind(msg) = message {
+            match msg {
+                message::StartInternalBitcoindMsg::ExePathEdited(exe_path) => {
+                    self.exe_path.value = exe_path;
+                    self.exe_path.valid = PathBuf::from(self.exe_path.value.clone()).exists();
+                }
+                message::StartInternalBitcoindMsg::Start => {
+                    let args = vec![
+                        format!("-chain={}", self.network.to_core_arg()),
+                        format!("-conf={}", self.conf_path.to_string_lossy()),
+                    ];
+                    // TODO: How to run in Windows as daemon-like process?
+                    let res = std::process::Command::new(self.exe_path.value.clone())
+                        .args(&args)
+                        .output();
+                    self.started = Some(res);
+                }
+            };
+        };
+        Command::none()
+    }
+
+    fn apply(&mut self, _ctx: &mut Context) -> bool {
+        if let Some(res) = self.started.as_ref() {
+            // This only tells us the command was executed successfully.
+            // Connection is checked in the following `DefineBitcoind` step.
+            return res.is_ok();
+        }
+        false
```

Comment:
By using `-daemonwait` i think we could get rid of the "set up connection to the Bitcoin node" step that follows:
![image](https://github.com/wizardsardine/liana/assets/22457751/048ae4bb-e9be-4c69-bf1d-c8d5040e2897)

It would be a good thing because it's either the user is managing their bitcoind and they need to give us the details to be able to connect, or we do and they shouldn't be bothered (they wouldn't even know the details).

Reply:
I've changed to `daemonwait` in https://github.com/wizardsardine/liana/commit/2a28dc72e063f867f6eba70dbb30b6265bccd7e8, 

If I'm no longer waiting for the output when starting bitcoind (as per https://github.com/wizardsardine/liana/pull/592#discussion_r1288317633), is using `daemon`/`daemonwait` now redundant?

Reply:
Yes. As suggested in https://github.com/wizardsardine/liana/pull/592#discussion_r1288536749 let's stick to running-a-bitcoind-in-foreground-in-a-thread for now. We can make it a daemon in a future release.